src/routeGeneration/tsoa-route.ts

```diff
@@ -13,20 +13,34 @@ export namespace TsoaRoute {
     [name: string]: ModelSchema;
   }
 
-  export interface ModelSchema {
-    enums?: string[] | number[];
-    properties?: { [name: string]: PropertySchema };
+  /**
+   * This is a convenience type so you can check .properties on the items in the Record without having TypeScript throw a compiler error. That's because this Record can't have enums in it. If you want that, then just use the base interface
+   */
+  export interface RefObjectModels extends TsoaRoute.Models {
+    [refNames: string]: TsoaRoute.RefObjectModelSchema;
+  }
+
+  export interface RefEnumModelSchema {
+    dataType: 'refEnum';
+    enums: string[] | number[];
+  }
+
+  export interface RefObjectModelSchema {
+    dataType: 'refObject';
+    properties: { [name: string]: PropertySchema };
     additionalProperties?: boolean | PropertySchema;
```

Comment:
```suggestion
     additionalProperties: boolean | PropertySchema;
```

I know this will cause TS to complain, but fixing the complaints might be worth the additional safety, what do you think?.

Reply:
I understand where you’re coming from, but Swagger doesn’t require that field. So I personally feel like that makes debugging things harder.